cpp/src/io/parquet/parquet_gpu.hpp

```diff
@@ -215,20 +215,25 @@ struct ColumnChunkDesc {
 /**
  * @brief Struct describing an encoder column
  */
-struct EncColumnDesc : stats_column_desc {
+struct parquet_column_device_view : stats_column_desc {
   uint32_t *dict_index;    //!< Dictionary index [row]
   uint32_t *dict_data;     //!< Dictionary data (unique row indices)
   uint8_t physical_type;   //!< physical data type
   uint8_t converted_type;  //!< logical data type
   // TODO (dm): Evaluate if this is sufficient. At 4 bits, this allows a maximum 16 level nesting
```

Comment:
Should we re-evaluate this now? Is it really worth the bit operations and the 16 nesting depth limit to smash this together into `level_bits`?

Reply:
Just saw this after merging. But the comment is wrong. 16 is not the nesting depth but the bit size of the nesting depth, which aligns with parquet apache reference impl. Our limit is slightly lower at 8 bits (max 255 level nesting) because the algorithm would have a really bad performance for large nesting.

I agree this is an over optimization. I'll remove it the next time I touch it.